platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrClientFactory.java

```diff
@@ -30,5 +31,7 @@
    * @return the newly created {@code SolrClient}
    * @throws IllegalArgumentException if <code>core</code> is <code>null</code>
    */
-  public SolrClient newClient(String core);
+  SolrClient newClient(String core);
+
+  SolrClient newClient(String collection, Map<String, Object> properties);
```

Comment:
❓ Is this flavor of `newClient` function only getting used in the split provider? **Besides** the usages that are just impl updates for the interfaces? If so, I think downstream can wrap the `SolrClient` with a proxy object that adds the enhanced functionality without extending this API.